stages/org.osbuild.ostree.deploy

```diff
@@ -84,12 +85,33 @@ SCHEMA_2 = """
 },
 "inputs": {
   "type": "object",
-  "required": ["commits"],
   "additionalProperties": false,
+  "oneOf": [{
+    "required": ["commits"]
+  }, {
+    "required": ["images"]
+  }, {
+      "not": { 
+        "anyOf": [{
+            "required": ["commits"]
+          }, {
+            "required": ["images"]
+          }] 
+      }
+  }],
```

Comment:
I'm trying to understand the `not/anyOf` part. Can you dumb it down for me?

Reply:
When I didn't put in any inputs (so that it matches the old example using the `org.osbuild.ostree.pull` stage), I got a schema error since it didn't match either having a `commits` or `images` input. This is the logic for the three statements you've highlighted:

```
# Case 1: We have a commits input
"required": ["commits"] = True
"required": ["images"] = False
"not/anyOf" = ! ( True || False ) = False

# Case 2: We have an images input
"required": ["commits"] = False
"required": ["images"] = True
"not/anyOf" = ! ( True || False ) = False

# Case 3: We have neither a commits or images input
"required": ["commits"] = False
"required": ["images"] = False
"not/anyOf" = ! ( False || False ) = True
```
As seen above, each of the cases only results in one True statement which is needed to pass the `oneOf` statement (basically an XOR).

Reply:
In addition, we want it to fail if we specify more than one input as well:
```
# Case 4: We have both a commits or images input
"required": ["commits"] = True
"required": ["images"] = True
"not/anyOf" = ! ( True || True ) = False
```
Which fails the XOR since it results in two `True` statements

Reply:
👍